src/test/java/com/zimbra/cert/GetDomainCertTest.java

```diff
@@ -0,0 +1,136 @@
+  }
+
+  @Test
+  public void shouldReturnSuccessForDomainWithValidCert() throws Exception {
+    final Domain domain = mock(Domain.class);
+    when(provisioning.get(DomainBy.id, domainId)).thenReturn(domain);
+    when(domain.getDomainName()).thenReturn("domainName");
+    when(domain.getSSLCertificate())
+        .thenReturn(
+                "-----BEGIN CERTIFICATE-----\n"
+                + "MIIFPjCCBCagAwIBAgISAyeF5ryS59TmpV5xTYquviaNMA0GCSqGSIb3DQEBCwUA\n"
+                + "MDIxCzAJBgNVBAYTAlVTMRYwFAYDVQQKEw1MZXQncyBFbmNyeXB0MQswCQYDVQQD\n"
+                + "EwJSMzAeFw0yMjExMjIxNjI2MDFaFw0yMzAyMjAxNjI2MDBaMCcxJTAjBgNVBAMT\n"
+                + "HHdlYm1haWwtYWNtZS5kZW1vLnpleHRyYXMuaW8wggEiMA0GCSqGSIb3DQEBAQUA\n"
+                + "A4IBDwAwggEKAoIBAQDdo/18KgJWki8yGVKC+cSV1wUJ9RKuOzLS7FdHd0RIt0H6\n"
+                + "lT9UriqWd2Yoc49wGJnvs2JY9K8oS3qj1VCAMMUrLBPt6NXPrPAU9uCG7p4e1R8o\n"
+                + "jgJq1Wumhxaua+btaPnkEJefGge0zuNDP0aEusK+zH9mCxgMqTDcbkjqaI6FFxFb\n"
+                + "++HTMe/c4quAput2FzpmUY5loLsomcd4eq28pWP4zRSli31HxEBFNgk6V6zVKAGl\n"
+                + "d92jrYNXNzNVQE3zAa5Ds9VyJq4Z6/8/Irl9lG7GKeF9c4TIBzPUvoHQOcbyC3LU\n"
+                + "1mlJtbkHiouGwFkSgJGH0NPmZzj8eA5VNhEjm+GVAgMBAAGjggJXMIICUzAOBgNV\n"
+                + "HQ8BAf8EBAMCBaAwHQYDVR0lBBYwFAYIKwYBBQUHAwEGCCsGAQUFBwMCMAwGA1Ud\n"
+                + "EwEB/wQCMAAwHQYDVR0OBBYEFGHHL5z1LwZqLDE1YHc76YWctfCuMB8GA1UdIwQY\n"
+                + "MBaAFBQusxe3WFbLrlAJQOYfr52LFMLGMFUGCCsGAQUFBwEBBEkwRzAhBggrBgEF\n"
+                + "BQcwAYYVaHR0cDovL3IzLm8ubGVuY3Iub3JnMCIGCCsGAQUFBzAChhZodHRwOi8v\n"
+                + "cjMuaS5sZW5jci5vcmcvMCcGA1UdEQQgMB6CHHdlYm1haWwtYWNtZS5kZW1vLnpl\n"
+                + "eHRyYXMuaW8wTAYDVR0gBEUwQzAIBgZngQwBAgEwNwYLKwYBBAGC3xMBAQEwKDAm\n"
+                + "BggrBgEFBQcCARYaaHR0cDovL2Nwcy5sZXRzZW5jcnlwdC5vcmcwggEEBgorBgEE\n"
+                + "AdZ5AgQCBIH1BIHyAPAAdwB6MoxU2LcttiDqOOBSHumEFnAyE4VNO9IrwTpXo1Lr\n"
+                + "UgAAAYSgYBDwAAAEAwBIMEYCIQDCZVS/DDy0C2R9IrY1ScTonqeedlI7ZjXo7R5R\n"
+                + "ZunbhwIhAMwCpmEsi0NUx1PhhNUwP3x85bRUEq4S+w/0/Nwj1I0tAHUAtz77JN+c\n"
+                + "Tbp18jnFulj0bF38Qs96nzXEnh0JgSXttJkAAAGEoGAS9AAABAMARjBEAiAbZ6Ag\n"
+                + "fsdche4bRXR6G/CAfNk6sRyA5W3ThLWyn9+vfAIgdYMflSb3bZyQbuhKB4zt3s1O\n"
+                + "BXrSDRXXrPYnF5rMWhEwDQYJKoZIhvcNAQELBQADggEBAJXIkzjX3thAZyBSj9lr\n"
+                + "It8ipidZNghgI27d7cqntkdsIDWQYyRRbS05GObMoAEiQnFti8EGnqQqEFkgOAj1\n"
+                + "9wXjSQdOGJjzm+giMhci2VMBC4N/7fWOSxWQ+jmVsYOLUFFb8VBuhWpwU5VHOAcv\n"
+                + "6cHuQKNOw8cYm9BFtdKqeexcTkflpWVsNK/CxIiK3cf9/p43MLO73AdGPT2g6DEk\n"
+                + "f+6Ay6pRLopibLZgGWqqBxevG9Pag5VN1CXcR5cu431Ic5l6NLJp58ED9qw8rf/7\n"
+                + "2gAhU3H9TC8UfXXnezzd5PSRjFAi05ciYyUQv+jtAN/U/wGDpiZ1QySvq0p6NC0J\n"
+                + "f+g=\n"
+                + "-----END CERTIFICATE-----\n");
+
+    final GetDomainCert getDomainCert = new DumbGetDomainCertHandler();
```

Comment:
I suggest to find another way to mock this behavior, because:
- we are not using the class that we want to test
- it's not clear what the "dumb" class is effectively doing (I know that checkDomainRight output is ignored, but still). What if we override another method of the class in future?

Consider using Mockito spy() to make it more obvious what we are doing in the test (@Polpetta I know you're not a fan of spying, maybe you can suggest an alternative) or check if it is possible to give the user the Right mocking zcs.

Reply:
I checked and to mock zcs we would have to go deep downmocking the Provisioning Config class (not worth it).
Also the method is protected, it would have been nice if the Authorization class was provided instead of extending everything. I just don't like wrapping in another class. 
Leave it as is or instead define an Anonymous so we don't encourage using DumbGetDomainCertHandler class.

Reply:
> Leave it as is or instead define an Anonymous so we don't encourage using DumbGetDomainCertHandler class.

i've made it to be an inner class